lib/auth/password_test.go

```diff
@@ -237,37 +239,54 @@ func TestServer_ChangePassword(t *testing.T) {
 	password := mfa.Password
 
 	tests := []struct {
-		name    string
-		newPass string
-		device  *TestDevice
+		name                        string
+		oldPass                     string
+		newPass                     string
+		userVerificationRequirement string
+		device                      *TestDevice
 	}{
 		{
-			name:    "OK TOTP-based change",
-			newPass: "llamasarecool11",
-			device:  mfa.TOTPDev,
+			name:                        "OK TOTP-based change",
+			oldPass:                     password,
+			newPass:                     "llamasarecool11",
+			userVerificationRequirement: "discouraged",
+			device:                      mfa.TOTPDev,
 		},
 		{
-			name:    "OK Webauthn-based change",
-			newPass: "llamasarecool13",
-			device:  mfa.WebDev,
+			name:                        "OK Webauthn-based change",
+			oldPass:                     "llamasarecool11",
+			newPass:                     "llamasarecool13",
+			userVerificationRequirement: "discouraged",
+			device:                      mfa.WebDev,
+		},
+		{
+			name:                        "OK passwordless change",
+			oldPass:                     "",
+			newPass:                     "llamasarecool15",
+			userVerificationRequirement: "required",
+			device:                      mfa.PasswordlessDev,
 		},
 	}
 
 	authServer := srv.Auth()
-	ctx := context.Background()
+	ctx := authz.ContextWithUser(
+		context.Background(),
+		authz.LocalUser{Username: username, Identity: tlsca.Identity{Username: username}},
```

Comment:
I would rather use a client for this user than mess with the context's internal variables.

Try this:

```go
	userClient, err := srv.NewClient(TestUser(username))
```

and then use `userClient` for the user->Auth interactions.

Reply:
Yeah, will do. I thought that there's something important in the fact that we're using the server directly, since the test's name mentions the word "server" explicitly.